selfservice/hook/require_verified_address.go

```diff
@@ -0,0 +1,139 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package hook
+
+import (
+	"net/http"
+
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/verification"
+	"github.com/ory/kratos/text"
+	"github.com/ory/kratos/x"
+	"github.com/ory/kratos/x/nosurfx"
+	"github.com/ory/x/otelx"
+	"github.com/ory/x/otelx/semconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/kratos/ui/node"
+
+	"github.com/ory/herodot"
+
+	"github.com/ory/kratos/identity"
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/session"
+)
+
+var _ login.PostHookExecutor = new(AddressVerifier)
+
+type (
+	addressVerifierDependencies interface {
+		config.Provider
+		nosurfx.CSRFTokenGeneratorProvider
+		nosurfx.CSRFProvider
+		verification.StrategyProvider
+		verification.FlowPersistenceProvider
+		identity.PrivilegedPoolProvider
+		x.WriterProvider
+		x.TracingProvider
+	}
+	AddressVerifier struct {
+		r addressVerifierDependencies
+	}
+)
+
+func NewAddressVerifier(r addressVerifierDependencies) *AddressVerifier {
+	return &AddressVerifier{
+		r: r,
+	}
+}
+
+func (e *AddressVerifier) ExecuteLoginPostHook(w http.ResponseWriter, r *http.Request, _ node.UiNodeGroup, f *login.Flow, s *session.Session) (err error) {
+	ctx, span := e.r.Tracer(r.Context()).Tracer().Start(r.Context(), "selfservice.hook.Verifier.do")
+	r = r.WithContext(ctx)
+	defer otelx.End(span, &err)
+
+	// TODO remove once flag is removed.
+	if e.r.Config().UseLegacyRequireVerifiedLoginError(ctx) {
+		if f.Active != identity.CredentialsTypePassword {
```

Comment:
why only non-password flows?

Reply:
I don't know - it's only for the legacy flag, where this was the case. So I kept it there as I don't want to change behavior for clients that have the flag on!